### ISSUE TITLE
Add ability to close date picker on iOS using VoiceOver without selecting a date

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#633]
# Feature Request

## Description
Currently, you can close the date picker on mobile by tapping outside of the date picker when it is open. It would be nice to provide and explicit close button when the date picker is displayed in overlay mode that would be more accessible to VoiceOver on iOS

**Comment by bjankord** [comment]
Thinking of about adding a specific button the date picker popup to close it. Sighted users can click off of the date picker to close the popup, but we don't currently have a good solution for people who are unable to see the screen to be able to close the popup. Making an explicit close button in the popup would help with this. The remaining question I have is whether or not that button should be visible.

Will review with @scottwilmarth and see if we can find a pragmatic solution for this.

**Comment by bjankord** [comment]
Tested with a visually-hidden button within the popup and this works well with VoiceOver on iOS. Tested in other browsers and didn't notice any issues with adding this visually-hidden button. There is a parent container with tabIndex="-1" we use in combination with focus-trap-react. With adding this button, we can remove that tabIndex and rely on the close button being the first interactive element within the popup and we can shift focus to it when the date picker popup is open.

**Comment by bjankord** [comment]
Related branch: https://github.com/cerner/terra-framework/tree/date-picker-close-button
